docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -0,0 +1,416 @@
+In this design, scheduling will be globally optimal and the resource usage rate will be good. There is no scheduling conflict/interference. But scalability won't be good since each Req is essentially handled serially. The allocation rate will not be good either. It also introduces a lot of traffic to the control plane network. Implementation is also complicated since a) a Global Arbitrator needs to be implemented. b) two-phase commit needs to be implemented, i.e. each scheduler needs to reserve the resource from its selected node and then either commit the resource or releases the resource after receiving a final decision from the Global Arbitrator.
+![image.png](/images/1.3.5.png)
+
+### 1.3.6 Two-Level Scheduler
+
+Scheduling is divided into two levels: one first-level scheduler/manager and multiple second-level cluster schedulers. The first-level scheduler selects the best cluster
+and the selected cluster level scheduler selects the best node.
+
+In this design, scheduling could be globally optimal and resource usage rate can be good. There is no scheduling conflict/interference. But scalability and allocation rate won't be good since all the requests go through a single first-level global scheduler. The first-level global scheduler can become a bottleneck.
+![image.png](/images/1.3.6.png)
+
+There is a variation to this architecture---the first-level scheduler/manager "selects" the best cluster by offering to compute resources to multiple parallels, independent second-level schedulers. That is, the first-level scheduler/manager decides how many resources to offer each cluster scheduler based on some organizational policy, such as fair sharing or priority, while cluster schedulers decide whether to accept the resource and which VMs/Containers to run on them. This two-level scheduling architecture appears to provide ﬂexibility and parallelism, but in practice, the scalability is limited by the capacity of the first-level scheduler/manager and the locking algorithm. The offer-reject cycle introduces extra latency, which impacts the allocation rate.
+
+### 1.3.7 Summary of Scheduler Design Pain Points
+
+Global scheduling must meet several goals simultaneously:
+
+- High scalability to support a high volume of incoming scheduling requests.
+- Rapid decision making and low scheduling latency to support a high allocation rate
+- Globally optimal scheduling to support high resource utilization across all cloud data centers and cloud edge sites
+- Satisfying the user-supplied placement constraints
+
+These goals usually conflict with each other. As analyzed in the previous sessions, each type of scheduler architecture only meets a few of the above goals, but not all the goals.
+
+# 2. Global Scheduler Overall Architecture Design
+
+## 2.1 Key Requirement Analysis
```

Comment:
Whether to add batch scheduling requirements
Unlike Kubernetes, you can apply for a VM or container at a time. Multiple resources can be applied for at a time. All resources can be successfully scheduled. Otherwise, the scheduling fails and no VM is created
I think this may be a key technology and requirement

Reply:
Yes, I will add that in.